CONTRIBUTING.md

```diff
@@ -31,15 +33,31 @@ It is intended to be migrated to Jekyll markdown and to be published with github
 
 ## Contributing code
 
+Before development is started, a [JIRA task or bug](#working-with-jira) should have been created for the issue.
 All development is managed using [pull requests](#pull-requests-workflow).
-[Commits](#commits) must be in logical, consistent units and have a good commit message.
+[Commits](#commits) must be in logical, consistent units and have a good commit message that includes the JIRA key of the corresponding issue.
 Every commit must carry a Signed-off-by tag to assert your agreement with the [DCO](#developers-certificate-of-origin).
 Pull requests are reviewed and checks are performed on them before they are merged.
 Code should adhere to the [coding style](#coding-style).
 
 Note that everybody follows these same guidelines.
 Core contributors, occasional contributors and maintainers are all alike.
 
+### Working with JIRA
+
+Every bug or feature needs to have a JIRA task associated with it, so that we can track the progress and see what people are working on.
+An exception is made for "hotfixes": small fixes that are uncontroversial, e.g. fixing whitespace in one file. Those don't need an explicit JIRA issue.
+
+To enable automatic tracking between JIRA and git, we ask that all branches (except for hotfixes) contain the JIRA id, and all commits contain it as well. This is unfortunately necessary for matching up the information between the two systems. Adding the JIRA key to pull/merge request names is also recommended, but not required.
+
+The JIRA issue should be created before any work is started. When work is started, the developer should transition the issue to "In progress" and assign his- or herself to the issue.
```

Comment:
```suggestion
The JIRA issue should be created before any work is started.
When work is started, the developer should transition the issue to "In progress" and assign his- or herself to the issue.
```
(more of these issues below, I'm not going to fix all of them :-)

Reply:
Done, thanks